test/spec/proxySpec.js

```diff
@@ -0,0 +1,1656 @@
+if (typeof window === 'undefined') {
+    var jsdom = require("jsdom").jsdom;
+    var doc = jsdom(undefined, undefined);
+    global.window = doc.defaultView;
+    global.document = doc.defaultView.document;
+}
+
+/* we need json-patch-duplex for jsonpatch.compare */
+if (typeof jsonpatch === 'undefined' || !jsonpatch.compare /* if jsonpatch is loaded without compare, we need to load duplex */) {
+    jsonpatch = require('./../../src/json-patch-duplex.js');
+}
+if (typeof JsonObserver === 'undefined') {
+    JsonObserver = require('./../../src/json-observe').default;
+}
+if (typeof _ === 'undefined') {
+    var _ = require('underscore');
+}
+function trigger(eventName, elem) {
+    if (typeof document !== 'undefined') {
+        fireEvent((elem || document.documentElement), eventName);
+    }
+}
+
+function getPatchesUsingGenerate(objFactory, objChanger) {
+    var obj = objFactory();
+    var jsonObserver = new JsonObserver(obj);
+    var observedObj = jsonObserver.observe(true);
+    objChanger(observedObj);
+    return jsonObserver.generate();
+}
+
+function getPatchesUsingCompare(objFactory, objChanger) {
+    var obj = objFactory();
+    var mirror = JSON.parse(JSON.stringify(obj));
+    objChanger(obj);
+    return jsonpatch.compare(mirror, JSON.parse(JSON.stringify(obj)));
+}
+
+//http://stackoverflow.com/questions/827716/emulate-clicking-a-link-with-javascript-that-works-with-ie
+function fireEvent(obj, evt) {
+    var fireOnThis = obj;
+    if (document.createEvent) {
+        var evObj = document.createEvent(evt.indexOf('mouse') > -1 ? 'MouseEvents' : 'KeyboardEvent');
+        evObj.initEvent(evt, true, false);
+        fireOnThis.dispatchEvent(evObj);
+
+    } else if (document.createEventObject) {
+        var evObj = document.createEventObject();
+        fireOnThis.fireEvent('on' + evt, evObj);
+    }
+}
+
+var customMatchers = {
+    /**
+     * This matcher is only needed in Chrome 28 (Chrome 28 cannot successfully compare observed objects immediately after they have been changed. Chrome 30 is unaffected)
+     * @param obj
+     * @returns {boolean}
+     */
+    toEqualInJson: function (util, customEqualityTesters) {
+        return {
+            compare: function (actual, expected) {
+                return {
+                    pass: JSON.stringify(actual) == JSON.stringify(expected)
+                }
+            }
+        }
+    },
+    toReallyEqual: function (util, customEqualityTesters) {
+        return {
+            compare: function (actual, expected) {
+                return {
+                    pass: _.isEqual(actual, expected)
+                }
+            }
+        }
+    }
+};
+
+describe("duplex", function () {
+
+    beforeEach(function () {
+        jasmine.addMatchers(customMatchers);
+    });
+
+    describe("toReallyEqual", function () {
```

Comment:
Are we testing just custom matcher here? if so I'd move it out of `describe('proxy')` to `describe('custom matchers')